airflow/dag_processing/manager.py

```diff
@@ -777,8 +777,9 @@ def clear_nonexistent_import_errors(self, session):
         :param session: session for ORM operations
         """
         query = session.query(errors.ImportError)
-        if self._file_paths:
-            query = query.filter(~errors.ImportError.filename.in_(self._file_paths))
+        files = list_py_file_paths(self._dag_directory, include_examples=False, include_zip_paths=True)
```

Comment:
Sadly, `list_py_file_paths` is an expensive operation...

Reply:
Do we have an idea how to do it better? I have not looked at all the processing paths that are involved, but I am not sure how "expensive" this is - seems to be done once per loop and something that we have to do anyway periodically (as we cannot really rely on inotify kind of API due to various filesystems DAG folder can be on.

Maybe this check should happen independently on another theread in-parallel to parsing ? Any other ideas @ephraimbuddy ?

Reply:
I think there are two possible alternatives. One is to introduce a new attribute on DagFileProcessorManager that stores the “full” paths, so we can use it instead of `_file_paths` here. The other is to introduce a new column on ImportError that store the filesystem path (i.e. path to the zip file) so we can filter it against `_file_paths`.

The root issue here is that both `_file_paths` and `ImportError.filename` essentially has double meaning—they both represent the actual filesystem entry (path to an actual file), and a Python code loading target (path for the interpreter). Right now `_file_paths` is a list of filesystem entries, while `ImportError.filename` is a code target, and trying to comparing them is fundamentally not going to work.

Reply:
> I think there are two possible alternatives. One is to introduce a new attribute on DagFileProcessorManager that stores the “full” paths, so we can use it instead of `_file_paths` here. The other is to introduce a new column on ImportError that store the filesystem path (i.e. path to the zip file) so we can filter it against `_file_paths`.
> 
> The root issue here is that both `_file_paths` and `ImportError.filename` essentially has double meaning—they both represent the actual filesystem entry (path to an actual file), and a Python code loading target (path for the interpreter). Right now `_file_paths` is a list of filesystem entries, while `ImportError.filename` is a code target, and trying to comparing them is fundamentally not a good idea.

Agree. I think having filesystem_path in import errors is a good idea - and likely it's an easy one that can be automatically set on migration, so should be rather easy to do.

Reply:
> Agree. I think having filesystem_path in import errors is a good idea - and likely it's an easy one that can be automatically set on migration, so should be rather easy to do.

I like the idea too